### PR TITLE
Fix #6 Now you can also return value from define function like can be done with require.js

### LINKED ISSUE #6: Amd module with return doesn't get populated on node.js
A module like this will return {}

```
define(function(require, exports, module){

    var logic = {
        a: 3
    }
    return logic;
});
```

Here is the index.js

```
var myRequire2 = require('enhanced-require')(module);
console.log(myRequire2('./test'))
```

This will output `{}` instead of `{logic: 3}`. Why is this happening and how to fix it?

I can't change all the `define(function(require, exports, module){` files to make them work with `module.exports = logic`. (The module.exports works ok)

On the client side with webpack works perfectly fine.

Thanks

### PR DESCRIPTION
**Pull request by bitplanets**
For example

```
define(function(require, exports, module){

    var logic = {
        a: 3
    }
    return logic;
});
```

will now work.

But this will continue to work too

```
define(function(require, exports, module){

    var logic = {
        a: 3
    }
    exports.a= logic.a;
});
```

### CODE REVIEW

**Comment by bitplanets** [comment]
@sokra Please merge this and add publish to npm. Thanks

**Comment by sokra** [comment]
Doesn't it make more sense to check if the return value `!== undefined` instead of testing for an empty object.

Please write a test case.

**Comment by bitplanets** [comment]
No because is easier for some function to return something by mistake, than set a property on exports variable. 
Ok, I will try to make a test (:

**Comment by bitplanets** [comment]
@sokra Added the tests

**Comment by sokra** [comment]
Could add a test that a empty module returns an empty object:

``` js
define(function(require, exports, module){
});
```

should be `{}`

**Comment by bitplanets** [comment]
@sokra  Added, also allows to return falsy values.

File: lib/RequireContext.js [hunk 1]
```diff
@@ -251,10 +251,17 @@ RequireContext.prototype.theDefine = function theDefine(dependencies, fn, arg3)
...
-			fn(reqFn, parent.exports, parent);
-		else
+		if(typeof fn == "function"){
+			var fnReturnExports = fn(reqFn, parent.exports, parent);
+			// If nothing has been added to the parent.exports then replace with the
+			// return value of the function.
+			if(Object.keys(parent.exports).length === 0){
+                // You can return 0, false and other falsy values.
+                parent.exports = fnReturnExports === void 0 ? {} : fnReturnExports;
+			}
```
Comment (sokra) [inline lib/RequireContext.js]:
``` js
var orignalExports = parent.exports;
var fnReturnExports = fn(reqFn, parent.exports, parent);
if(orignalExports === parent.exports && Object.keys(orignalExports).length === 0 && fnReturnExports !== undefined) {
  parent.exports = fnReturnExports;
}
```

File: test/commonjs.js [hunk 2]
```diff
@@ -77,4 +77,29 @@ describe("commonjs", function() {
...
+        data.should.be.a("object");
+        done();
+    });
+
+    it("should be able to return a falsy value", function(done) {
+        var data = req('./fixtures/commonjs/wrappedFalsy');
+        data.should.be.equal(false);
+        done();
+    });
+
```
Comment (sokra) [inline test/commonjs.js:103]: And tests for these modules:

``` js
define(function(require, exports, module) {
  function Test() {}
  module.exports = new Test();
});

define(function(require, exports, module) {
  module.exports = undefined;
});
```

**Comment by bitplanets** [comment]
Ok, I've made the changes.
